### ISSUE TITLE
Customising the Auth view using AWSMobileClient

### ISSUE THREAD

**Issue by jamesingham** [aws-amplify/aws-sdk-ios#1113]
I have just updated my project to use use AWSMobileClient (2.7.1). I am able to sign in using credentials for a Cognito User Pools user successfully like so:
```
AWSMobileClient.sharedInstance().showSignIn(navigationController: self.navigationController!, { (userState, error) in
    // ...
}
```

When I run this and the user is authenticated, I am then able to call `AWSMobileClient.sharedInstance().getTokens()`.

However I need to be able to customise the login view. So I found a section in the [documentation](https://aws-amplify.github.io/docs/ios/authentication#customization) which suggests to do this:
```
AWSMobileClient.sharedInstance()
    .showSignInScreen(navigationController: self.navigationController!,
                      signInUIOptions: SignInUIOptions(
                            canCancel: false,
                            logoImage: UIImage(named: "MyCustomLogo"),
                            backgroundColor: UIColor.black)) { (result, err) in
                            //handle results and errors               
}
```

I'm not sure if the documents are out of date or something, but after following the suggested fixes in Xcode (the method has been renamed, and has three variables in the callback now), I end up with this:
```
AWSMobileClient.sharedInstance()
            .showSign(inScreen: self.navigationController!, signInUIConfiguration: SignInUIOptions(canCancel: false, logoImage: UIImage(named: "MyCustomLogo"), backgroundColor: UIColor.black)) { (signInProviderKey, signInProviderToken, error) in
                //handle results and errors
        }
```

Because `showSign()` is now returning a provider key and token I am assuming that I then need to call `AWSMobileClient.sharedInstance().federatedSignIn()` with those values. This seems to work (signing the user in) however I still cannot call `AWSMobileClient.sharedInstance().getTokens()` after calling `federatedSignIn()`. The interesting thing is if do this to login as a user and then re-run the application, it will auto login as I expect and then `AWSMobileClient.sharedInstance().getTokens()` works just fine!

So I'm just confused as to why it doesn't work when first logging in by calling `showSign()` then `federatedSignIn()`. If I go back to using just `showSignIn()` as explained above it all works in both scenarios.

There is also a minor issue that the logo does not set to my custom image. All other options in `SignInUIOptions` works as expected.

**Environment:**
 - SDK Version: 2.7.1
 - Dependency Manager: Cocoapods
 - Swift Version: 4.2

**Device Information:**
 - Device: iPhone XS, Simulator
 - iOS Version: iOS 12.1

**Comment by rohandubal** [comment]
Hello @jamesingham 

Sorry that you are having issues using `AWSMobileClient`. The issue which you were facing regarding customization of UI is fixed in the documentation. The API was incorrectly documented.

The correct code(which should be updated in documentation) is:

```swift
AWSMobileClient.sharedInstance()
                            .showSignIn(navigationController: self.navigationController!,
                                        signInUIOptions: SignInUIOptions(canCancel: false,
                                                                         logoImage: UIImage(named: "MyCustomLogo"),
                                                                         backgroundColor: UIColor.black)) { (result, err) in
                                                                            
}
```

Regarding the federation, if you use the above API, the federation is done automatically behind the scenes and you do not have to call the `federatedSignIn` API. You can just call `getAWSCredentials` to get the federated credentials.

Please let us know if you still face issues.

Thanks,
Rohan

**Comment by edwin0527** [comment]
Hello @rohandubal,

This was helpful, but when I use the following code:

`AWSMobileClient.sharedInstance()
                            .showSignIn(navigationController: self.navigationController!,
                                        signInUIOptions: SignInUIOptions(canCancel: false,
                                                                         logoImage: UIImage(named: "MyCustomLogo"),
                                                                         backgroundColor: UIColor.black)) { (result, err) in
                                                                            
}`

The "MyCustomLogo" does not appear and I still see the default logo.  However, the background does change to black, but I'm more concern with changing the logo. I am pretty new to iOS development so I could be missing something simple but I just can figure out why the logo is not changing.

**Comment by jamesingham** [comment]
> Regarding the federation, if you use the above API, the federation is done automatically behind the scenes and you do not have to call the `federatedSignIn` API. You can just call `getAWSCredentials` to get the federated credentials.

Thanks for that. I can confirm it's working.

However as mentioned by @edwin0527, setting `logoImage` `UIImage` doesn't actually change the image in the sign in view.

**Comment by rohandubal** [comment]
Thanks for reporting this @jamesingham 

I will try to reproduce this on my side and update back.

**Comment by rohandubal** [comment]
Hello, this is will be fixed in future SDK release.

**Comment by rohandubal** [comment]
Hello,

This issue is resolved in `2.7.4` SDK release. Please let us know if you continue to face issues.

Thanks,
Rohan
